processor/tailsamplingprocessor/internal/sampling/ottl.go

```
@@ -0,0 +1,124 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspanevent"
+)
+
+type ottlConditionFilter struct {
+	sampleSpanExpr      expr.BoolExpr[ottlspan.TransformContext]
+	sampleSpanEventExpr expr.BoolExpr[ottlspanevent.TransformContext]
+	ErrorMode           ottl.ErrorMode
+	logger              *zap.Logger
+}
+
+var _ PolicyEvaluator = (*ottlConditionFilter)(nil)
+
+// NewOTTLConditionFilter looks at the trace data and returns a corresponding SamplingDecision.
+func NewOTTLConditionFilter(settings component.TelemetrySettings, spanConditions, spanEventConditions []string, errMode ottl.ErrorMode) (PolicyEvaluator, error) {
+	filter := &ottlConditionFilter{
+		ErrorMode: errMode,
+		logger:    settings.Logger,
+	}
+
+	var err error
+	if len(spanConditions) > 0 {
+		if filter.sampleSpanExpr, err = filterottl.NewBoolExprForSpan(spanConditions, filterottl.StandardSpanFuncs(), ottl.PropagateError, settings); err != nil {
+			return nil, err
+		}
+	}
+
+	if len(spanEventConditions) > 0 {
+		if filter.sampleSpanEventExpr, err = filterottl.NewBoolExprForSpanEvent(spanEventConditions, filterottl.StandardSpanEventFuncs(), ottl.PropagateError, settings); err != nil {
```

We should use errorMode here instead of hard-coding propagate.

---

Fixed.